autoemulate/experimental/tuner.py

```diff
@@ -52,25 +59,32 @@ def run(self, model_class: type[Emulator]) -> tuple[list[float], list[ModelConfi
         model_config_tested: list[ModelConfig] = []
         val_scores: list[float] = []
 
-        for _ in range(self.n_iter):
+        for _ in tqdm(range(self.n_iter)):
             # randomly sample hyperparameters and instantiate model
             model_config: ModelConfig = {
                 k: v[np.random.randint(len(v))] for k, v in tune_config.items()
             }
-            if isinstance(model_class, gpytorch.models.ExactGP):
-                m = model_class(train_x, train_y, **model_config)
-                assert isinstance(m, Emulator)
+
+            if issubclass(model_class, gpytorch.models.ExactGP):
+                m = model_class(
+                    train_x,
+                    train_y,
+                    likelihood=MultitaskGaussianLikelihood(
+                        num_tasks=tuple(train_y.shape)[1]
+                    ),
+                    **model_config,
+                )
                 m.fit(train_x, train_y)
             else:
                 m = model_class(**model_config)
-                assert isinstance(m, Emulator)
                 # TODO: check if pass as dataloader
                 m.fit(train_x, train_y)
```

Comment:
We might consider refactoring this to something like:
```python
model_class(train_x, train_y, model_config)
```
for all classes and add an init to the `Emulator` base class so that the initialize handling is not required by the tuner. This also relates to how data is passed to the compare, tuner, model etc.

Reply:
Added a revision to enable the init and fit to be the same for different models ([d354062](https://github.com/alan-turing-institute/autoemulate/pull/384/commits/d35406270bdd1ea08fb1f10f03eeb62dca1a3ae3))

Reply:
I like this

Reply:
Thanks!